vm/devices/storage/disk_nvme/nvme_driver/fuzz/fuzz_nvme_driver.rs

```diff
@@ -111,14 +124,17 @@ impl FuzzNvmeDriver {
                 let buf_range = OwnedRequestBuffers::linear(0, 16384, true); // TODO: [use-arbitrary-input]
                 self.namespace
                     .read(
-                        target_cpu,
+                        target_cpu % self.cpu_count,
```

Comment:
Instead of sanitizing inputs in the fuzzer, it's generally considered better practice to validate the inputs in the thing being targeted (the driver) and return an error. I understand that this is a purely internal interface today, but that doesn't guarantee it will remain that way forever into the future.

Reply:
I would tend to agree with this rule of thumb. However, these interfaces are not the actual target of this fuzzing. The target is the interaction between the `nvme_driver` and the underlying nvme device.

So, two things:

1. If we find ourselves doing any sort of sanitization in `fuzz_emulated_device.rs`, then I agree this is cause for concern: `fuzz_emulated_device.rs` is the set of interfaces that `nvme_driver` will call that we'd like to target with fuzz data, and
2. We need a mechanism to say "this is invalid input that represents a programming error. So, `panic!` unless we're running under a fuzzer, and if we're running under a fuzzer then bail gracefully." I'm also happy to further discuss how we handle contract violations for private interfaces, but we should include @jstarks in that conversation; he has previously expressed that he'd rather see the nvme driver code `assert!` and `panic!` for illegal input.
3. Should we promote `nvme_driver` outside the `disk_nvme` crate, it should be rather straightforward to remove the sanitization of this fuzzer (and these type of contract bugs will be hit quite quickly).

Reply:
1. In that case, can we add a top level comment to each module clarifying what each is targeting, whether sanitization is permitted or not, etc?
2. This could potentially be done with `cfg(fuzzing)`, but I agree it's probably worth a more in-depth discussion of these sorts of scenarios as a hole first.
3. Perhaps we should comment or otherwise document each such location then? Just to make it that much more obvious.

Reply:
Regarding the second item (`cfg(fuzzing)`) - I was mulling over this yesterday and think that's where we'll end up. I agree we want this and had already asked @gurasinghMS to come up with a solution (tracked as one of the bullet points in #521). I'm motivated to get past these fuzz failures before the holidays, and I also do not want to take that task away from him.

Regarding the others - also good feedback; I'll update this PR to address.